diff --git a/src/20251126_AaveV2Ethereum_WindingDownLendMigrationContract/LendToAaveMigrator/src/contracts/Original b/src/20251126_AaveV2Ethereum_WindingDownLendMigrationContract/LendToAaveMigrator/src/contracts/LendToAaveMigrator.sol
index 59db969..1a669c8 100644
--- a/src/20251126_AaveV2Ethereum_WindingDownLendMigrationContract/LendToAaveMigrator/src/contracts/Original
+++ b/src/20251126_AaveV2Ethereum_WindingDownLendMigrationContract/LendToAaveMigrator/src/contracts/LendToAaveMigrator.sol
@@ -1,8 +1,9 @@
 // SPDX-License-Identifier: agpl-3.0
 pragma solidity ^0.8.0;
 
-import {IERC20} from 'solidity-utils/contracts/oz-common/interfaces/IERC20.sol';
+import {IERC20} from 'openzeppelin-contracts/contracts/token/ERC20/IERC20.sol';
 import {VersionedInitializable} from './dependencies/upgradeability/VersionedInitializable.sol';
+import {MiscEthereum} from 'aave-address-book/MiscEthereum.sol';
 
 /**
  * @title LendToAaveMigrator
@@ -10,10 +11,15 @@ import {VersionedInitializable} from './dependencies/upgradeability/VersionedIni
  * @author Aave
  */
 contract LendToAaveMigrator is VersionedInitializable {
+  error MigrationFinished();
+  error MigrationNotFinished();
+
+  uint256 public constant MIGRATION_END_TIMESTAMP = 1767225600; // January 1, 2026, UTC
+
   IERC20 public immutable AAVE;
   IERC20 public immutable LEND;
   uint256 public immutable LEND_AAVE_RATIO;
-  uint256 public constant REVISION = 2;
+  uint256 public constant REVISION = 3;
 
   uint256 public _totalLendMigrated;
 
@@ -37,7 +43,7 @@ contract LendToAaveMigrator is VersionedInitializable {
    * @param lend the address of the LEND token
    * @param lendAaveRatio the exchange rate between LEND and AAVE
    */
-  constructor(IERC20 aave, IERC20 lend, uint256 lendAaveRatio) public {
+  constructor(IERC20 aave, IERC20 lend, uint256 lendAaveRatio) {
     AAVE = aave;
     LEND = lend;
     LEND_AAVE_RATIO = lendAaveRatio;
@@ -46,43 +52,17 @@ contract LendToAaveMigrator is VersionedInitializable {
   }
 
   /**
-   * @dev initializes the implementation and rescues the LEND sent to the contract
-   * by migrating them to AAVE and sending them to the AaveMerkleDistributor
-   * and then burning the LEND tokens
-   * @param aaveMerkleDistributor address of the AAVE rescue distributor
-   * @param lendToMigratorAmount amount of lend sent to migrator that need to be rescued
-   * @param lendToLendAmount amount of lend sent to LEND that need to be rescued
-   * @param lendToAaveAmount amount of lend sent to AAVE that need to be rescued
+   * @dev empty initializer
+   */
+  function initialize() public initializer {}
+
+  /**
+   * The DAO voted on stopping the migration on 1st of January 2026
+   * Once that time has passed, all remaining AAVE tokens can be claimed to the ecosystem reserve.
    */
-  function initialize(
-    address aaveMerkleDistributor,
-    uint256 lendToMigratorAmount,
-    uint256 lendToLendAmount,
-    uint256 lendToAaveAmount
-  ) public initializer {
-    uint256 lendAmount = lendToMigratorAmount + lendToLendAmount + lendToAaveAmount;
-    uint256 migratorLendBalance = _totalLendMigrated + lendToMigratorAmount;
-
-    // account for the LEND sent to the contract for the total migration
-    _totalLendMigrated += lendAmount;
-
-    // transfer AAVE + LEND sent to this contract
-    uint256 amountToRescue = lendAmount / LEND_AAVE_RATIO;
-    AAVE.transfer(aaveMerkleDistributor, amountToRescue);
-
-    LEND.transfer(address(LEND), migratorLendBalance);
-
-    emit LendMigrated(address(this), lendAmount);
-    emit AaveTokensRescued(address(this), aaveMerkleDistributor, amountToRescue);
-
-    // checks that the amount of AAVE not migrated is less or equal as the amount of AAVE disposable for migration
-    // we have found that there was a previous small surplus on the AAVE token amount found on the LendToAaveMigrator
-    // contract previous to the rescue, that is why we need to use <= instead of == . This amount is 582968318731898974 (0,58 AAVE)
-    require(
-      (LEND.totalSupply() - LEND.balanceOf(address(LEND)) - lendToAaveAmount) / LEND_AAVE_RATIO <=
-        AAVE.balanceOf(address(this)),
-      'INCORRECT_BALANCE_RESCUED'
-    );
+  function transferRemainingFundsToEcosystemReserve() external {
+    require(block.timestamp > MIGRATION_END_TIMESTAMP, MigrationNotFinished());
+    AAVE.transfer(address(MiscEthereum.ECOSYSTEM_RESERVE), AAVE.balanceOf(address(this)));
   }
 
   /**
@@ -99,6 +79,7 @@ contract LendToAaveMigrator is VersionedInitializable {
    * @param amount the amount of LEND to be migrated
    */
   function migrateFromLEND(uint256 amount) external {
+    require(block.timestamp <= MIGRATION_END_TIMESTAMP, MigrationFinished());
     require(lastInitializedRevision != 0, 'MIGRATION_NOT_STARTED');
 
     _totalLendMigrated = _totalLendMigrated + amount;
